crates/uv-distribution-types/src/prioritized_distribution.rs

```diff
@@ -634,37 +636,290 @@ impl IncompatibleWheel {
+                    arch
+                } else if let Some(arch) = tag.strip_prefix("musllinux_") {
+                    // Skip over the version tags (e.g., given `musllinux_1_2`, skip over `1` and `2`).
+                    let mut parts = arch.splitn(3, '_');
+                    if parts
+                        .next()
+                        .and_then(|part| part.parse::<u16>().ok())
+                        .is_none()
+                    {
+                        debug!("Failed to parse musllinux major version from tag: {tag}");
+                        continue;
+                    };
+                    if parts
+                        .next()
+                        .and_then(|part| part.parse::<u16>().ok())
+                        .is_none()
+                    {
+                        debug!("Failed to parse musllinux minor version from tag: {tag}");
+                        continue;
+                    };
+                    let Some(arch) = parts.next() else {
+                        debug!("Failed to parse musllinux architecture from tag: {tag}");
+                        continue;
+                    };
+                    arch
+                } else if let Some(arch) = tag.strip_prefix("manylinux_") {
+                    // Skip over the version tags (e.g., given `manylinux_2_17`, skip over `2` and `17`).
+                    let mut parts = arch.splitn(3, '_');
+                    if parts
+                        .next()
+                        .and_then(|part| part.parse::<u16>().ok())
+                        .is_none()
+                    {
+                        debug!("Failed to parse manylinux major version from tag: {tag}");
+                        continue;
+                    };
+                    if parts
+                        .next()
+                        .and_then(|part| part.parse::<u16>().ok())
+                        .is_none()
+                    {
+                        debug!("Failed to parse manylinux minor version from tag: {tag}");
+                        continue;
+                    };
+                    let Some(arch) = parts.next() else {
+                        debug!("Failed to parse manylinux architecture from tag: {tag}");
+                        continue;
+                    };
+                    arch
+                } else {
+                    continue;
+                };
+                tag_marker.and(MarkerTree::expression(MarkerExpression::String {
+                    key: MarkerValueString::PlatformMachine,
+                    operator: MarkerOperator::Equal,
+                    value: arch.to_string(),
                 }));
+
+                marker.or(tag_marker);
             }
-            _ => {}
         }
     }
     marker
 }
+
+#[cfg(test)]
+mod tests {
+    use std::str::FromStr;
+
+    use super::*;
+
+    #[test]
+    fn test_implied_markers() {
+        macro_rules! assert_markers {
```

Comment:
Can this be a regular function/closure?

Reply:
It can... but then you don't get proper line numbers when you hit an assertion failure :(